app/models/miq_request_workflow.rb

```diff
@@ -855,20 +855,28 @@ def ems_has_clusters?
     false
   end
 
+  FOLDER_PATH_SEPARATOR  = " / ".freeze
+  DATACENTER_FOLDER_TYPE = "Datacenter".freeze
   def get_ems_folders(folder, dh = {}, full_path = "")
+    path = full_path
     if folder.evm_object_class == :EmsFolder
       if folder.hidden
         return dh if folder.name != 'vm'
       else
-        full_path += full_path.blank? ? folder.name.to_s : " / #{folder.name}"
-        dh[folder.id] = full_path unless folder.type == "Datacenter"
+        path = full_path.dup
+        if path.blank?
+          path << folder.name.to_s
+        else
+          path << FOLDER_PATH_SEPARATOR << folder.name
```

Comment:
@NickLaMuro - Since there isn't a ternary here constantly generating new `String` objects. Within the confines of a conditional Is the `CONSTANT` still more performant than just including the "/" here?

Reply:
I want to say "yes", but I would rather have some proof.  Let me run some tests quick.

Reply:
Ended up doing by testing against the `DATACENTER_FOLDER_TYPE` constant and putting my findings there:

https://github.com/ManageIQ/manageiq/pull/17358/files#r188783333

but the same should apply here as well.

Reply:
So my tendency is to go the easier to read path, which would mean pulling out the constants and replacing them with `string_name.freeze`.

Reply:
Mmmk, I can go with that.  I tend to do constants myself because I can wrap my brain around it a little easier.  When it is in the `"string".freeze` mode, I still think of/read it as "allocate string, then make it immutable", when it really isn't doing the "allocate string" bit at all.

Any who, I will make the change.  👍